### PR TITLE
[CI]updated drone.star file

### PR DESCRIPTION
**Pull request by sushmita56**
## Description
This is an update on drones.star file 
- using drone-cancel-previous-builds
- using thegeeklab/drone-github-comment instead of jmccann/drone-github-comment for github comments 
- moving docker image references to constants
- using caches images
- Do not run PHP unit tests with Oracle by default

## Related Issue
- Part of issue: 
- https://github.com/owncloud/QA/issues/700
- https://github.com/owncloud/QA/issues/717
- https://github.com/owncloud/QA/issues/729
- https://github.com/owncloud/QA/issues/728
- https://github.com/owncloud/core/issues/38679

### CODE REVIEW

**Comment by phil-davis** [comment]
There are still a lot of diffs from what is in the activity app master branch. Please copy all the code from there - we want it to be exactly the same in each oC10 app.

File: .drone.star [hunk 1]
```diff
@@ -2005,7 +1995,7 @@ def githubComment(earlyFail):
     if (earlyFail):
         return [{
             "name": "github-comment",
-            "image": "jmccann/drone-github-comment:1",
+            "image": THEGEEKLAB_DRONE_GITHUB_COMMENT,
             "pull": "if-not-exists",
             "settings": {
                 "message_file": "%s/comments.file" % dir["base"],
```
Comment (kiranparajuli589) [inline .drone.star]: this new image has a little different configuration. please refer to the changes in the activity app and fix the case here

**Review by kiranparajuli589 — CHANGES_REQUESTED** [review]
also add `.idea` dir to the `gitignore`

File: .drone.star [hunk 2]
```diff
@@ -1,7 +1,33 @@
+BANST_AWS_CLI = "banst/awscli"
+DRONE_CI_ALPINE = "drone/cli:alpine"
+MAILHOG = "mailhog/mailhog"
```
Comment (amrita-shrestha) [inline .drone.star]:
```suggestion
MAILHOG_MAILHOG = "mailhog/mailhog"
```

File: .drone.star [hunk 3]
```diff
@@ -1,7 +1,33 @@
+BANST_AWS_CLI = "banst/awscli"
+DRONE_CI_ALPINE = "drone/cli:alpine"
```
Comment (amrita-shrestha) [inline .drone.star]:
```suggestion
DRONE_CLI = "drone/cli:alpine"
```

File: .drone.star [hunk 4]
```diff
@@ -1,7 +1,33 @@
...
+OC_CI_CORE = "owncloudci/core"
+OC_CI_DRONE_CANCEL_PREVIOUS_BUILDS = "owncloudci/drone-cancel-previous-builds"
+OC_CI_DRONE_SKIP_PIPELINE = "owncloudci/drone-skip-pipeline"
+OC_CI_NODEJS = "owncloudci/nodejs:%s"
+OC_CI_ORACLE_XE = "owncloudci/oracle-xe:latest"
+OC_CI_PHP = "owncloudci/php:%s"
 OC_CI_WAIT_FOR = "owncloudci/wait-for:latest"
-OC_CI_NODEJS = "owncloudci/nodejs:14"
+OC_OPS_ELASTIC_SEARCH = "owncloudops/elasticsearch:%s"
+OC_SCALITY_S3_SERVER = "owncloudci/scality-s3server"
```
Comment (amrita-shrestha) [inline .drone.star]:
```suggestion
OC_CLI_SCALITY_S3_SERVER = "owncloudci/scality-s3server"
```

**Review by amrita-shrestha — DISMISSED** [review]

File: .drone.star [hunk 5]
```diff
@@ -1,7 +1,33 @@
+BANST_AWS_CLI = "banst/awscli"
+DRONE_CI_ALPINE = "drone/cli:alpine"
+MAILHOG = "mailhog/mailhog"
+MINIO_MC_RELEASE_2020_VERSION = "minio/mc:RELEASE.2020-12-18T10-53-53Z"
```
Comment (kiranparajuli589) [inline .drone.star]:
```suggestion
MINIO_MC = "minio/mc:RELEASE.2020-12-18T10-53-53Z"
```

File: .drone.star [hunk 6]
```diff
@@ -574,15 +571,15 @@ def javascript(ctx, withCoverage):
             "base": dir["base"],
             "path": "server/apps/%s" % ctx.repo.name,
         },
-        "steps": installCore(ctx, "daily-master-qa", "sqlite", False) +
+        "steps": skipIfUnchanged(ctx, "lint") +
```
Comment (kiranparajuli589) [inline .drone.star]:
```suggestion
        "steps": skipIfUnchanged(ctx, "unit-tests") +
```

File: .drone.star [hunk 7]
```diff
@@ -746,7 +773,8 @@ def phpTests(ctx, testType, withCoverage):
                         "base": dir["base"],
                         "path": "server/apps/%s" % ctx.repo.name,
                     },
-                    "steps": installCore(ctx, "daily-master-qa", db, False) +
+                    "steps": skipIfUnchanged(ctx, "lint") +
```
Comment (kiranparajuli589) [inline .drone.star]:
```suggestion
                    "steps": skipIfUnchanged(ctx, "unit-tests") +
```

File: .drone.star [hunk 8]
```diff
@@ -1077,8 +1100,9 @@ def acceptance(ctx):
                         "base": dir["base"],
                         "path": "testrunner/apps/%s" % ctx.repo.name,
                     },
-                    "steps": installCore(ctx, testConfig["server"], testConfig["database"], testConfig["useBundledApp"]) +
-                             installTestrunner(ctx, "7.4", testConfig["useBundledApp"]) +
+                    "steps": skipIfUnchanged(ctx, "lint") +
```
Comment (kiranparajuli589) [inline .drone.star]:
```suggestion
                    "steps": skipIfUnchanged(ctx, "acceptance-tests") +
```

File: .drone.star [hunk 9]
```diff
@@ -1186,14 +1209,14 @@ def sonarAnalysis(ctx, phpVersion = "7.4"):
                          ],
                      },
                  ] +
+                 skipIfUnchanged(ctx, "lint") +
```
Comment (kiranparajuli589) [inline .drone.star]:
```suggestion
                 skipIfUnchanged(ctx, "unit-tests") +
```

**Review by kiranparajuli589 — DISMISSED** [review]

**Review by phil-davis — APPROVED** [review]

**Pull request merged**